keywrap/pkcs11/keywrapper_pkcs11_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+   Copyright The ocicrypt Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package pkcs11
+
+import (
+	"os"
+	"testing"
+
+	"github.com/containers/ocicrypt/config"
+	"github.com/containers/ocicrypt/crypto/pkcs11"
+	"github.com/containers/ocicrypt/utils/softhsm"
+)
+
+var (
+	SOFTHSM_SETUP = "../../scripts/softhsm_setup"
+)
+
```

Comment:
Is the teardown related to state outside the program? Should teardown be run each time as part of `TestMain`?

Reply:
All softhsm_setup operations are global. Ideally there would initially be the startup, then test cases can call the getpubkey, and at the end there's the teardown.

Reply:
So I could only find that `TestMain` has package-wide behavior. This would work if all our softhsm related operations were done in the same package, but this is not the case. What we would need is a global setup function running before all tests (in all directories) and a teardown at the very end once all tests (from all directories)  have finished...